packages/tokens/src/rwa/identity_storage_registry/storage.rs

```diff
@@ -440,57 +415,63 @@ pub fn modify_country_profile(e: &Env, account: &Address, index: u32, profile: &
 /// Using this function in public-facing methods may create significant security
 /// risks as it could allow unauthorized modifications.
 pub fn delete_country_profile(e: &Env, account: &Address, index: u32) {
-    let profile_to_remove = get_country_profile(e, account, index);
+    let mut profiles = get_country_profiles(e, account);
 
-    let count = get_country_profile_count(e, account);
-
-    // Can't overflow because `get_country_profile` would panic if count == 0
-    let last_index = count - 1;
-    // Revert if no CountryProfile is left
-    if last_index == 0 {
+    if profiles.len() == 1 {
         panic_with_error!(e, IRSError::EmptyCountryProfiles)
     }
 
-    // If the profile to be deleted is not the last one,
-    // move the last profile into its place to keep the list compact.
-    if index != last_index {
-        let last_key = IRSStorageKey::CPEnumerable(CPEnumerableKey {
-            account: account.clone(),
-            index: last_index,
-        });
-        let last_profile: CountryProfile = e
-            .storage()
-            .persistent()
-            .get(&last_key)
-            .unwrap_or_else(|| panic_with_error!(&e, IRSError::CountryProfileNotFound));
-
-        let current_key =
-            IRSStorageKey::CPEnumerable(CPEnumerableKey { account: account.clone(), index });
-        e.storage().persistent().set(&current_key, &last_profile);
-    }
+    let profile_to_remove = profiles
+        .get(index)
+        .unwrap_or_else(|| panic_with_error!(e, IRSError::CountryProfileNotFound));
 
-    // Remove the last profile's storage entry
-    let key_to_remove = IRSStorageKey::CPEnumerable(CPEnumerableKey {
-        account: account.clone(),
-        index: last_index,
-    });
-    e.storage().persistent().remove(&key_to_remove);
+    profiles.remove(index);
 
-    // Decrement the count
-    e.storage().persistent().set(&IRSStorageKey::CPCount(account.clone()), &(count - 1));
+    let key = IRSStorageKey::CountryProfiles(account.clone());
+    e.storage().persistent().set(&key, &profiles);
 
     emit_country_profile_event(e, CountryProfileEvent::Removed, account, &profile_to_remove);
 }
 
-/// Low-level function that tries to retrieve a persistent storage value and
+/// Helper function that tries to retrieve a persistent storage value and
 /// extend its TTL if the entry exists.
 ///
 /// # Arguments
 ///
-/// * `e` - The environment reference.
+/// * `e` - The Soroban reference.
 /// * `key` - The key required to retrieve the underlying storage.
 fn get_persistent_entry<T: TryFromVal<Env, Val>>(e: &Env, key: &IRSStorageKey) -> Option<T> {
     e.storage().persistent().get::<_, T>(key).inspect(|_| {
         e.storage().persistent().extend_ttl(key, IDENTITY_TTL_THRESHOLD, IDENTITY_EXTEND_AMOUNT);
     })
 }
+
+/// Helper function to Add multiple country profiles without chaching whether
+/// identity already exists and without event emitting.
```

Comment:
it also doesn't check whether the country profiles already exist for the account from what I understand. Should we also add this detail to the inline docs?